---
title: "edgeR and DESeq2 Immunotherapy Dataset"
author: "Andreea Murariu"
date: "`r Sys.Date()`"
output: html_document
---

```{r}
#Installation of Packages edgeR, DESeq2, and ALDEx2
library(edgeR,quietly=T) 
library(DESeq2,quietly=T)
library(ALDEx2,quietly=T)
```

1. Using edgeR to identify differentially expressed genes (DEGs) in the Immunotherapy Dataset 
```{r}
#Setting the Working Directory
#andreea setting
## gg may comment this out 
setwd("/Users/andreeamurariu/Documents/github/3383 Project/")
getwd() 
```

** Using series matrix file **

```{r}
readCount<-read.table(file="data/GSE91061_raw_counts_GRCh38.p13_NCBI.tsv", header = T, skip=35, sep='\t', row.names = 1)
#pull out just the accessino numbers and patient identifiers and put them in another file
d <- read.table(file="data/metadata.txt", header=F, row.names=1, sep='\t')
```

```{r}
conditions <- rep("Pre", 109)
grep("_On",d)
# troubleshooting check
#conditions[grep("_On",d)] <- "On"
```


```{r}
#COUNT MATRIX PREPROCESSING 
conds <- factor(conditions)
y <- DGEList(counts=readCount, group=conds) 

# use edgeR filtering for rows as SOP
keep<-filterByExpr(y) 
# y <- y[keep,,keep.lib.sizes=FALSE] 
# 
# design <- model.matrix(~conds)
# fit <- glmQLFit(y, design)
# qlf <- glmQLFit(fit, coef=2) # returns error message because outputs negative values, which should not be possible in reality (ie. gene expression levels should not be negative values)
# 
# count_norm=cpm(y) 
# count_norm<-as.data.frame(count_norm) 
```
#edgeR will have to be omitted from the rest of the tests

2. Using DESeq2 to identify differentially expressed genes (DEGs) in the Immunotherapy Dataset 



3. Using ALDEx2 to identify differentially expressed genes (DEGs) in the Immunotherapy Dataset 
ALDEx2 t-test and Wilcoxon rank-sum test
```{r}
#use edgeR filtering to keep rows for analysis
read.keep <- readCount[keep,]
aldex.out <- aldex(read.keep, conditions, gamma=1e-3)

```
