---
title: "edgeR and DESeq2 Immunotherapy Dataset"
author: "Andreea Murariu"
date: "`r Sys.Date()`"
output: html_document
---

```{r}
#Installation of Packages edgeR, DESeq2, and ALDEx2
library(edgeR,quietly=T) 
library(DESeq2,quietly=T)
library(ALDEx2,quietly=T)
```

1. Using edgeR to identify differentially expressed genes (DEGs) in the Immunotherapy Dataset 
```{r}
#Setting the Working Directory
#andreea setting
## gg may comment this out 
setwd("/Users/andreeamurariu/Documents/github/3383-Project/")
getwd() 
```

** Using series matrix file **

```{r}
readCount<-read.table(file="data/GSE91061_raw_counts_GRCh38.p13_NCBI.tsv", header = T, skip=35, sep='\t', row.names = 1)
#pull out just the accession numbers and patient identifiers and put them in another file
d <- read.table(file="data/metadata.txt", header=F, row.names=1, sep='\t')
```

```{r}
conditions <- rep("Pre", 109)
grep("_On",d)
# troubleshooting check
#### GG
# This needed to be uncommented
###
conditions[grep("_On",d)] <- "On"
conds <- data.frame(conditions)
conds$condition <- factor(conds$condition)

```


```{r}
### I commented all this out as irrelevant except line 53
### still need the edgeR package
#COUNT MATRIX PREPROCESSING 
conds.edge <- factor(conditions)
y <- DGEList(counts=readCount, group=conds.edge) 

# use edgeR filtering for rows as SOP
keep<-filterByExpr(y) 
# y <- y[keep,,keep.lib.sizes=FALSE] 
# 
# design <- model.matrix(~conds)
# fit <- glmQLFit(y, design)
# qlf <- glmQLFit(fit, coef=2) # returns error message because outputs negative values, which should not be possible in reality (ie. gene expression levels should not be negative values)
# 
# count_norm=cpm(y) 
# count_norm<-as.data.frame(count_norm) 
```
#edgeR will have to be omitted from the rest of the tests

2. Using DESeq2 to identify differentially expressed genes (DEGs) in the Immunotherapy Dataset 

```{r}
#new trial of DESeq2 code

read.keep <- readCount[keep,]

#Creating a DESeqDataSet

dds <- DESeqDataSetFromMatrix(countData = read.keep,colData = conds, design= ~condition) #error message with saying unused arguments

#DESeq2 analysis
### GG comment this out
# deseq2out <- DESeq (dds, test=c("LRT"))
deseq2out <- DESeq (dds) # use the default test
res <- results(deseq2out)
plotMA(res) # show the MA plot
res


```
```{r}
#permute the immunotherapy dataset
eread.keepx=matrix(nrow=1000,ncol=109)
	set.seed(100)
	index=1
	for(i in 1:1000){
		#x=rep(NA,109)
		#x[sample(1:109,58)]="On"
		#x[is.na(x)]="Pre"
		read.keep[index,]=x
		index=index+1
	}
	write.table(read.keep,file="permuted1.csv",sep="\t", quote=F,row.names = F,col.names = F) 
	
	
	
	sample(colnames(read.keep))

```


```{r}
read.keep <- readCount[keep,]
conds <- factor(conditions)
trueDE <-Reduce(intersect,read.keep)

SE1<-SummarizedExperiment(assays=read.keep,metadata=d,rowRanges=(2,17237)) # running into issue here, says count matrix should be numberic and that it currently has mode: list
DESeqDataSet(SE1)
DESeq(DESeqDataSet(), test=c(LRT),fit=parametric) 


##is normalization here necessary for our purposes?
if(norm =="permprenorm"){
  # Permutation under H0
  trueDE_samplehalf <- sample(trueDE, length(trueDE)/2)
  readCount[!(rownames(readCount) %in% trueDE_samplehalf), ] <-
    t(apply(readCount[!(rownames(readCount) %in% trueDE_samplehalf), ], 1, sample))
}

if(norm !="nonorm"){
  # Normalization
  y <- DGEList(counts=readCount, group=condition)
  keep <- filterByExpr(y)
  y <- y[keep, keep.lib.sizes=TRUE]
  y <- calcNormFactors(y,method="TMM")
  count_norm <- cpm(y, log=TRUE)
}else{
  count_norm <- as.matrix(readCount)
}

if(norm == "permpostnorm" | norm == "nonorm"){
  # Permutation under H0
  trueDE_samplehalf <- sample(trueDE, length(trueDE)/2)
  count_norm[!(rownames(count_norm) %in% trueDE_samplehalf), ] <-
    t(apply(count_norm[!(rownames(count_norm) %in% trueDE_samplehalf), ], 1, sample))
}


```


```{r}
DESeq2::plotMA(count_norm,alpha=0.01) #is there another type of plot I should be using instead for DESeq2?
```


3. Using ALDEx2 to identify differentially expressed genes (DEGs) in the Immunotherapy Dataset 
ALDEx2 t-test and Wilcoxon rank-sum test
```{r}
#use edgeR filtering to keep rows for analysis
read.keep <- readCount[keep,]
#aldex.out <- aldex(read.keep, conditions, gamma=1e-3)
#aldex.0.5 <- aldex(read.keep, conditions, gamma=0.5)
aldex.1 <- aldex(read.keep, conditions, gamma=1) 
aldex.5 <- aldex(read.keep, conditions, gamma=5) 
#save(aldex.out, file="output/aldex.out.Rda")
#save(aldex.0.5, file="output/aldex.0.5.Rda")
#these have already been run + data is saved
#load(file="output/aldex.out.Rda")
#load(file="output/aldex.0.5.Rda")
#pdf("no-gamma.pdf", height=5, width=5)
#aldex.plot(aldex.out)

save(aldex.1, file="output/aldex.1.Rda")
save(aldex.5, file="output/aldex.5.Rda")
load(file="output/aldex.1.Rda")
load(file="output/aldex.5.Rda")
pdf("gamma.1.pdf", height=5, width=7)
aldex.plot(aldex.1)
pdf("gamma.5.pdf", height=5, width=7)
aldex.plot(aldex.5)

#5,10, 20 in each group
#do tests w 2 p values and effect-size cutoff of 1


#how to deal with ones significant w p values and not effect
#what is the proportion of the ones sig w p / effect

#for positive control, when subset, are the scale parameters consistent across diff subsets - should be

# condition vector and randomize it

conds <- c(rep("H",5),rep("B",5))
r.conds <-sample(conds)
```
